src/kwlib.sh

```diff
@@ -371,3 +371,38 @@ function exit_msg()
   complain "$msg"
   exit "$err"
 }
+
+# This function compresses a given path to a .tar.gz file
+#
+# @path_to_compress The directory to compress
+# @file_path Where the compressed directory will be stored
+# @compression_type compression program used
+# @flag How to display (or not) the command used
+function generate_tarball()
+{
+  local path_to_compress="$1"
+  local file_path="$2"
+  local compression_type="$3"
+  local flag="$4"
+  local ret
+  local cmd
+
+  if [[ ! -d "$path_to_compress" ]]; then
+    complain "$path_to_compress" 'does not exist'
+    exit 22 #EINVAL
+  fi
+
+  if [[ -n "$compression_type" ]]; then
+    compression_type="--$compression_type"
```

Comment:
Can you check if we can validate this compression flag? Maybe the `tar` command provides something; if not, we can try to compress a dummy file just for checking (I would prefer the first option).